flows/actions/base.go

```diff
@@ -139,30 +139,22 @@ func (a *BaseAction) resolveLabels(run flows.FlowRun, step flows.Step, reference
 	return labels, nil
 }
 
-// BaseMsgAction is our base action for message that generate message events
-type BaseMsgAction struct {
-	BaseAction
+// MsgAction is our mixin for an action that generates message events
+type MsgAction struct {
```

Comment:
golang doesn't really do mixins but not sure what else you'd call this. Two of our actions (`reply`, `send_msg`) need to share some message creating functionality, two of them (`send_msg`, `start_session`) need to share some contact/group/urn resolving functionality.

Reply:
I think my bias would be that we duplicate the variable declarations (ie the serialization format) for the subactions and just turn these into utility methods. I don't like serialization formats being done through encapsulation / inheritance, I'd rather those be explicit, and it seems simple / clearer to just have utility methods for the resolving. I would guess that is slightly more Go-ish in that they love repeating themselves.

Reply:
I'm fine with that - that's how resolveLabels / resolveGroups currently work already